ruff.toml

```diff
@@ -3,11 +3,32 @@ extend = "pyproject.toml"
 
 [lint]
 extend-select = [
+	# upstream
+ 
 	"C901",
 	"PERF401",
 	"W",
+ 
+ 	# Ensure modern type annotation syntax and best practices
+	# Not including those covered by type-checkers or exclusive to Python 3.11+
+	"FA", # flake8-future-annotations
+	"F404", # late-future-import
+	"PYI", # flake8-pyi
```

Comment:
This rule turns out to be noisy (PYI024 in particular). It requires re-writing all namedtuples to typing.NamedTuples and can't be fixed mechanically.